src/dparse/parser.d

```diff
@@ -3133,30 +3161,46 @@ class Parser
     }
 
     /**
-     * Parses an IdentifierList
+     * Parses an IdentifierList.
      *
      * $(GRAMMAR $(RULEDEF identifierList):
-     *     $(LITERAL Identifier) ($(LITERAL ',') $(LITERAL Identifier))*
-     *     ;)
+     *      $(RULE identifierOrTemplateInstance)
+     *    | $(RULE identifierOrTemplateInstance) $(LITERAL '.') $(RULE identifierList)
+     *    | $(RULE identifierOrTemplateInstance) $(LITERAL '[') $(RULE assignExpression) $(LITERAL ']') $(LITERAL '.') $(RULE identifierList)
```

Comment:
The language spec and I disagree on the topic of "IdentifierList" being only a list of identifiers.